clarity-ext-scripts/clarity_ext_scripts/covid/knm_sminet_service.py

```diff
@@ -0,0 +1,157 @@
+        referring_clinic_name = provider.patient["managingOrganization"]["display"]
+        referring_clinic_county = self.get_county_from_organization(
+            provider.organization)
+        referring_doctor = provider.service_request["resource"]["requester"]["display"]
+
+        # TODO: We don't have an address for the referring clinic
+        return ReferringClinic(referring_clinic_name, "",
+                               referring_clinic_county, Doctor(referring_doctor))
+
+
+    def map_gender(self, knm_gender):
+        """Returns gender in the format required by SmiNet"""
+        pass
+
+    def create_patient(self, provider):
+        """Creates a Patient object from KNM data"""
+
+        def patient_identifier():
+            try:
+                patient_identifier = provider.patient["identifier"]
+            except KeyError:
+                raise IntegrationError(
+                    "Missing field 'identifier' on the patient resource for {}".format(provider))
+
+            if len(patient_identifier) == 0:
+                raise IntegrationError(
+                    "Field 'identifier' is empty on patient resource for {}".format(provider))
+
+            try:
+                return patient_identifier[0]["value"]
+            except:
+                raise IntegrationError(
+                    "First entry in 'identifier' doesn't have a value key for {}".format(provider))
+
+        def patient_gender():
+            gender = provider.patient["gender"]
+            try:
+                return self.MAP_GENDER_FROM_KNM_TO_SMINET[gender]
+            except KeyError:
+                raise IntegrationError(
+                    "Unexpected value for key 'gender' ({}) for {}".format(gender, provider))
+
+        def patient_name():
+            """
+            The name of the patient. Returns an empty string if we don't have the expected data
+            """
+            name = provider.patient["name"]
+
+            if len(name) == 0:
+                return ""
+
+            name = name[0]
+
+            try:
+                return name["text"]
+            except KeyError:
+                return ""
+
+        return Patient(patient_identifier(),
+                       patient_gender(),
+                       patient_name(),
+                       None)
+
+    def get_sminet_export(self, sample):
+        """
+        Returns a SmiNetLabExport from a sample 
+
+        :sample: A KNMSampleAccessor object
+        """
+
+        provider = ServiceRequestProvider(
+            self.knm_client, sample.org_uri, sample.org_referral_code)
+
+        sample_info = self.create_sample_info(provider, sample)
+        print(sample_info)
+        reporting_doctor = Doctor("Lars Engstrand")
+        print(reporting_doctor)
+        referring_clinic = self.create_referring_clinic(provider)
+        print(referring_clinic)
+        patient = self.create_patient(provider)
+        lab_result = LabResult("C",
```

Comment:
Should the `create_scov2_positive_lab_result` method perhaps be used here?

Reply:
Overall it looks like maybe the methods from `clarity-ext-scripts/clarity_ext_scripts/covid/sminet_settings.py` could be used in this file? Which would also to some extend fix part of the problem with hard coding in this file that I mentioned above.

Reply:
Yes, I've pulled this stuff up now.